coldfront/core/project/templates/project/project_join_list.html

```diff
@@ -129,13 +129,39 @@ <h5>Make a New Request</h5>
               Join
             </a>
           {% else %}
-            <form action="{% url 'project-join' project.pk %}" method="post">
-              {% csrf_token %}
-              <button class="btn btn-primary" type="submit">
+            <div class="modal fade" id="reasonModal" tabindex="-1" role="dialog" aria-labelledby="reasonModalLabel" aria-hidden="true">
+            <div class="modal-dialog" role="document">
+                <form action="{% url 'project-join' project.pk %}" method="post">
+                    {% csrf_token %}
+                    <div class="modal-content">
+                        <div class="modal-header">
+                            <h5 class="modal-title">Join Project</h5>
+                            <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+                                <span aria-hidden="true">&times;</span>
+                            </button>
+                        </div>
+
```

Comment:
Let's add some explanatory text in this modal: "Please provide a short reason that will be reviewed by the project managers."

Also, I think a `textarea` is more appropriate.